File: test-2025.9.2/编程/mergesort.py
def MergeSort(shuzu:list,left:int,right:int,temp:list):
    if(left < right):
        middle = int((left + right)/2 - 0.5)
        MergeSort(shuzu,left,middle,temp)
        MergeSort(shuzu,middle+1,right,temp)
        pl = left
        pr = middle + 1
        while(pl <= middle and pr <= right):
            if(shuzu[pl] < shuzu[pr]):
                temp[pl - left + pr - (middle + 1)] = shuzu[pl]
                pl+=1
            else:
                temp[pl - left + pr - (middle + 1)] = shuzu[pr]
                pr+=1
        while(pl <= middle):
            temp[pl - left + right  - middle] = shuzu[pl]
            pl+=1
        while(pr <= right):
            temp[pr - left] = shuzu[pr]
            pr+=1
        for i in range(left,right+1):
            shuzu[i] = temp[i - left]

def UseMergeSort(shuzu:list):
    shuzu_length = len(shuzu)
    copy_shuzu = [num for num in shuzu]
    temp_shuzu = [0 for _ in range(shuzu_length)]
    MergeSort(copy_shuzu,0,shuzu_length-1,temp_shuzu)
    return copy_shuzu

def main():
    input_number = input()
    old =[int(number) for number in input().split(" ")]
    old = old[0:int(input_number)]
    new = UseMergeSort(old)
    print(" ".join(map(str,new)),end="")

if __name__ == "__main__":
    main()

File: test-2025.9.2/编程/mergesort.cpp
using namespace std;
#include <vector>
#include <iostream>


void MergeSort(int *shuzu, int left, int right,int*temp)
{
    if (left == right)
    {
        return;
    }
    int middle = (left + right) / 2;
    MergeSort(shuzu, left, middle,temp);
    MergeSort(shuzu, middle + 1, right,temp);
    int p1 = left;
    int p2 = middle + 1;
    while ((p1 <= middle) && (p2 <= right))
    {
        if (shuzu[p1] < shuzu[p2])
        {
            temp[p1 - left + p2 - middle - 1] = shuzu[p1];
            p1++;
        }
        else
        {
            temp[p1 - left + p2 - middle - 1] = shuzu[p2];
            p2++;
        }
    }
    while (p2 <= right)
    {
        temp[p2 - left] = shuzu[p2];
        p2++;
    }
    while (p1 <= middle)
    {
        temp[p1 - left + right - middle] = shuzu[p1];
        p1++;
    }
    for (int i = left; i <= right; i++)
    {
        shuzu[i] = temp[i - left];
    }
    return;
}

int main()
{
    int number;
    cin >> number;
    int* shuzu = new int[number];
    int* temp = new int[number];
    for(int i = 0;i<number;i++)
    {
        cin >> shuzu[i];
    }
    MergeSort(shuzu,0,number-1,temp);
    for(int i = 0;i < number;i++)
    {
        cout << shuzu[i];
        if(i != number-1)
        {
            cout << " ";
        }
    }
}